llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -4636,6 +4640,14 @@ LoopVectorizationPlanner::selectInterleaveCount(VPlan &Plan, ElementCount VF,
       any_of(Plan.getVectorLoopRegion()->getEntryBasicBlock()->phis(),
              IsaPred<VPReductionPHIRecipe>);
 
+  // FIXME: implement interleaving for FindLast transform correctly.
+  if (any_of(make_second_range(Legal->getReductionVars()),
+             [](const RecurrenceDescriptor &RdxDesc) {
+               return RecurrenceDescriptor::isFindLastRecurrenceKind(
+                   RdxDesc.getRecurrenceKind());
+             }))
+    return 1;
```

Comment:
I think we need to move this check earlier. (see the check a bit before the call to selectInterleaveCount that handles isSafeForAnyVectorwidth). We take the max of UserIC (set via flag or pragma) and the IC returned here, so we still may end up interleaving if the user requests it. Would be good to add a test with forced interleaving.